include/util/compat.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef OSRM_COMPAT_HPP
+#define OSRM_COMPAT_HPP
+
+#if defined(__ARM_ARCH_7A__)
```

Comment:
Fixes are not really related to ARM architecture but to a specific libstdc++ version of the used toolchain. I would suggest to make it as a patch and apply it before building with the specific toolchain. Otherwise the patch brings redefinition errors for newer libstc++
```
[  1%] Building CXX object CMakeFiles/ENGINE.dir/src/engine/polyline_compressor.cpp.o
In file included from /home/miha/foss/osrm-backend/src/engine/polyline_compressor.cpp:3:0:
/home/miha/foss/osrm-backend/include/util/compat.hpp: In function ‘double std::round(double)’:
/home/miha/foss/osrm-backend/include/util/compat.hpp:21:33: error: ‘double std::round(double)’ conflicts with a previous declaration
     inline double round(double d)
                                 ^
In file included from /usr/arm-linux-gnueabihf/include/math.h:83:0,
                 from /usr/arm-linux-gnueabihf/include/c++/5/cmath:44,
```